version.go

```diff
@@ -1,47 +1,35 @@
 package main
 
 import (
-	"fmt"
-	"net/url"
-	"os"
-
-	functions "github.com/funcy/functions_go"
+	"github.com/fnproject/cli/client"
+	fnclient "github.com/fnproject/fn_go/client/version"
 	"github.com/urfave/cli"
+	"fmt"
 )
 
 // Version of Fn CLI
 var Version = "0.4.11"
 
 func version() cli.Command {
-	r := versionCmd{VersionApi: functions.NewVersionApi()}
 	return cli.Command{
 		Name:   "version",
 		Usage:  "displays fn and functions daemon versions",
-		Action: r.version,
+		Action: versionCMD,
 	}
 }
 
-type versionCmd struct {
-	*functions.VersionApi
-}
-
-func (r *versionCmd) version(c *cli.Context) error {
-	apiURL := os.Getenv("API_URL")
-	if apiURL == "" {
-		apiURL = "http://localhost:8080"
-	}
-
-	u, err := url.Parse(apiURL)
-	if err != nil {
-		return err
-	}
-	r.Configuration.BasePath = u.String()
-
-	fmt.Println("Client version:", Version)
-	v, _, err := r.VersionGet()
+func versionCMD(c *cli.Context) error {
+	t, reg := client.GetTransportAndRegistry()
+	// dirty hack, swagger paths live under /v1
+	// version is also there, but it shouldn't
+	// dropping base path to get appropriate URL for request eventually
+	t.BasePath = ""
+	version_client := fnclient.New(t, reg)
+	v, err := version_client.GetVersion(nil)
 	if err != nil {
 		return err
 	}
-	fmt.Println("Server version", v.Version)
+	fmt.Println("Client version: ", Version)
```

Comment:
this will mean that if one can't communicate with the server they can't see the client version. for downloads and in general for users to check their cli version, it would be nice to not have to have a live fn. at a minimum, can we move this back before the check of the server version?

I would suggest instead of returning the error to possibly put a `?` in place of the server version here, vs. showing the user:

```sh
$ fn --version
Client version: 0.4.2
HTTP client timeout: unable to reach http://thefntoruleallfn.com:8080/version
exit code 1
```

muting the error is kind of meh, but it's the version endpoint, and they'll find out soon enough if they try to do anything else. would be nice imo to see:

```sh
$ fn --version
Client version: 0.4.2
Server version: ?
```

Reply:
addressed